src/amber/controller/helpers/render.cr

```diff
@@ -15,8 +35,10 @@ module Amber::Controller::Helpers
       {% else %}
         {{ short_path = folder.gsub(/^.+?(?:controllers|views)\//, "") }}
         {% if folder.id.ends_with?(".ecr") %}
+          __content_filename__ = "#{{{path}}}/#{{{short_path.gsub(/\/[^\.\/]+\.ecr/, "")}}}/#{{{filename}}}" # -- 
           %content = render_template("#{{{path}}}/#{{{short_path.gsub(/\/[^\.\/]+\.ecr/, "")}}}/#{{{filename}}}")
         {% else %}
+          __content_filename__ = "#{{{path}}}/#{{{short_path.gsub(/\_controller\.cr|\.cr/, "")}}}/#{{{filename}}}" # -- 
           %content = render_template("#{{{path}}}/#{{{short_path.gsub(/\_controller\.cr|\.cr/, "")}}}/#{{{filename}}}")
         {% end %}
       {% end %}
```

Comment:
I tried this and doesn't work, I think you remove these `__content_filename__` lines and add this:

```cr
macro render(template = nil, layout = true, partial = nil, path = "src/views", folder = __FILE__)
  __content_filename__ = __FILE__

  {% if !(template || partial) %}
  raise "Template or partial required!"
  {% end %}

  {{ filename = template || partial }}


  # Render Template and return content
  {% if filename.id.split("/").size > 1 %}
    %content = render_template("#{{{filename}}}", {{path}})
  {% else %}
    {{ short_path = folder.gsub(/^.+?(?:controllers|views)\//, "") }}
    {% if folder.id.ends_with?(".ecr") %}
      %content = render_template("#{{{path}}}/#{{{short_path.gsub(/\/[^\.\/]+\.ecr/, "")}}}/#{{{filename}}}")
    {% else %}
      %content = render_template("#{{{path}}}/#{{{short_path.gsub(/\_controller\.cr|\.cr/, "")}}}/#{{{filename}}}")
    {% end %}
  {% end %}

  # Render Layout
  {% if layout && !partial %}
  content = %content
  render_template("#{{{path}}}/layouts/#{{{layout.class_name == "StringLiteral" ? layout : LAYOUT}}}")
  {% else %}
  %content
  {% end %}
end
```

^^ This way works just fine :tada: Maybe @sdogruyol has some ideas here :wink:

Reply:
Well thats not works for me, i just edit yiled_content and add this line 
```
puts __content_filename__,  CONTENT_FOR_BLOCKS[{{key}}][0]
```

And,

when render homepage template the output is 
src/views/home/homepage.ecr - __content_filename__
src/views/home/homepage.ecr - CONTENT_FOR_BLOCKS[{{key}}][0]

when render login template the output is 
/home/ysr/***/***/***/***/src/controllers/auth_controller.cr - __content_filename__
src/views/auth/register.ecr  - CONTENT_FOR_BLOCKS[{{key}}][0]


can you try it again?

Reply:
Still testing, sorry for the delay :sweat_smile:

Reply:
BTW, this PR need to solve some merge conflicts and formatting issues :+1: